PhysicsTools/PatAlgos/plugins/LeptonUpdater.cc

```diff
@@ -32,16 +40,33 @@ namespace pat {
           edm::ParameterSetDescription desc;
           desc.add<edm::InputTag>("src")->setComment("Lepton collection");
           desc.add<edm::InputTag>("vertices")->setComment("Vertex collection");
-          if (typeid(T) == typeid(pat::Muon)) descriptions.add("muonsUpdated", desc);
-          else if (typeid(T) == typeid(pat::Electron)) descriptions.add("electronsUpdated", desc);
+          desc.add<bool>("computeMiniIso", false)->setComment("Recompute miniIsolation");
+          desc.addOptional<edm::InputTag>("pfCandsForMiniIso", edm::InputTag("packedPFCandidates"))->setComment("PackedCandidate collection used for miniIso");
+          if (typeid(T) == typeid(pat::Muon)) {
+            desc.addOptional<std::vector<double>>("miniIsoParams")->setComment("Parameters used for miniIso (as in PATMuonProducer)");
+            descriptions.add("muonsUpdated", desc);
+          } else if (typeid(T) == typeid(pat::Electron)) {
+            desc.addOptional<std::vector<double>>("miniIsoParamsB")->setComment("Parameters used for miniIso in the barrel (as in PATElectronProducer)");
+            desc.addOptional<std::vector<double>>("miniIsoParamsE")->setComment("Parameters used for miniIso in the endcap (as in PATElectronProducer)");
+            descriptions.add("electronsUpdated", desc);
+          }
       }
 
       void setDZ(T & lep, const reco::Vertex & pv) const {}
+        
+      void readMiniIsoParams(const edm::ParameterSet & iConfig) { 
+          miniIsoParams_[0] = iConfig.getParameter<std::vector<double> >("miniIsoParams");
+          if(miniIsoParams_[0].size() != 9) throw cms::Exception("ParameterError", "miniIsoParams must have exactly 9 elements.\n");
```

Comment:
Is this "9" carved on stone? Couldn't it be compared with something more configurable or accessible from the code itself? Or at least define some adequately named const that refers to it (here and in lines 90 and 91 below)?

Reply:
this is the same as in
https://github.com/cms-sw/cmssw/blob/master/PhysicsTools/PatAlgos/plugins/PATMuonProducer.cc#L136-L138
because the py config is cloned the same mechanism (vector of floats instead of named variables) is used here.

It would probably make sense to move from vector to named parameters for config readibility but it is not an issue of this PR that just uses an existing interface and setup.

Reply:
Well, if that "9" gets propagated to several other places it is not only a matter of readability, but rather to avoid the risk of letting it unnoticed once the parameter set gets modified in a possible future development...
However, I agree that this is not an issue of this PR